server/eclipse-project/src/main/java/us/freeandfair/corla/query/CastVoteRecordQueries.java

```diff
@@ -389,6 +392,43 @@ public static CastVoteRecord atPosition(final Tribute tribute) {
                       tribute.ballotPosition);
   }
 
+  /** select cast_vote_record where uri in :uris **/
+  public static List<CastVoteRecord> atPosition(final List<Tribute> tributes) {
+
+    if (tributes.isEmpty()) {
+      return new ArrayList();
+    }
+
+    final List<String> uris = tributes.stream()
+      .map(t -> t.uri())
+      .collect(Collectors.toList());
+
+    final Session s = Persistence.currentSession();
+    final Query q =
+      s.createQuery("select cvr from CastVoteRecord cvr " +
+                    " where uri in (:uris) ");
+
+    q.setParameter("uris", uris);
+
+    final List<CastVoteRecord> results = q.getResultList();
+
+    final Set<String> foundUris = results.stream()
+      .map(cvr -> (String)cvr.getUri())
+      .collect(Collectors.toSet());
+
+    final Set<CastVoteRecord> phantomRecords = tributes.stream()
+      .filter(distinctByKey((Tribute t) -> {return t.uri();}))
```

Comment:
no, the distinctByKey function is a neat trick that I found on the internet (I should have added a link in the comments) The function it returns only returns true on the first instance, false otherwise.

Reply:
I'm just talking about simplifying the lambda expression, not removing `distinctByKey`.

Reply:
oh, I see, yes.